### PR TITLE
Consensus tests based on channels

### PR DESCRIPTION
**Pull request by eadventurous**

### CODE REVIEW

File: iroha/src/kura.rs [hunk 1]
```diff
@@ -63,6 +63,10 @@ impl Kura {
             }
         }
     }
+
+    pub fn blocks(&self) -> &[ValidBlock] {
```
Comment (humb1t) [inline iroha/src/kura.rs:67]: Getters should be avoided.

File: iroha/src/lib.rs [hunk 2]
```diff
@@ -21,24 +21,25 @@ use crate::{
     peer::{Peer, PeerId},
     prelude::*,
     queue::Queue,
-    sumeragi::{Message, Sumeragi},
-    torii::Torii,
+    sumeragi::{Message as SumeragiMessage, Sumeragi},
```
Comment (humb1t) [inline iroha/src/lib.rs:24]: I think `sumeragi::Message` is enough descriptive, why do wee need `SumeragiMessage`?

File: iroha/src/torii.rs [hunk 3]
```diff
@@ -38,12 +38,21 @@ impl Torii {
         Network::listen(Arc::new(RwLock::new(state)), url, handle_connection).await?;
         Ok(())
     }
+
+    pub async fn send(message: Message) -> Result<Response, String> {
```
Comment (humb1t) [inline iroha/src/torii.rs:42]: Why we need a duplicate of the `Network::send_request_to`?

File: iroha/src/sumeragi.rs [hunk 4]
```diff
@@ -276,14 +302,66 @@ impl Debug for Sumeragi {
...
-    use async_std::sync;
+    use crate::{crypto, ToriiMessageReceiver};
+    use async_std::{
+        prelude::*,
+        sync::{self, RwLock},
+    };
+    use std::{collections::HashMap, convert::TryFrom, time::Duration};
+
+    #[derive(Default)]
+    struct Router {
```
Comment (humb1t) [inline iroha/src/sumeragi.rs:313]: Please add the description or link to paper

File: iroha/src/sumeragi.rs [hunk 5]
```diff
@@ -108,28 +110,32 @@ impl Sumeragi {
...
+            let torii_sender = self.messages_to_torii_sender.write().await;
             for transaction in &transactions {
-                send_futures.push(Network::send_request_to(
-                    &self.leader().address,
-                    Request::new(
-                        uri::INSTRUCTIONS_URI.to_string(),
-                        RequestedTransaction::from(transaction).into(),
-                    ),
-                ));
+                torii_sender
```
Comment (humb1t) [inline iroha/src/sumeragi.rs:129]: Torii handles incoming requests, but doesn't need to be an abstraction about outgoing network requests.

File: iroha/src/sumeragi.rs (hunk 4 above)
Comment (eadventurous) [reply iroha/src/sumeragi.rs:313]: It's basically a network simulation through channels. It let's us not depend on torii in tests.

Comment (eadventurous) [reply iroha/src/sumeragi.rs:313]: The structure is for tests only.

**Pull request closed without merging**